.github/workflows/docker.yml

```diff
@@ -1,53 +1,126 @@
-name: Build & Push
+name: Push Docker Images
 # Build & Push builds the finschia docker image on every tag push
 
 on:
-  pull_request:
-    branches:
-      - main
   push:
     tags:
       - "v[0-9]+.[0-9]+.[0-9]+" # Push events to matching v*, i.e. v1.0, v20.15.10
       - "v[0-9]+.[0-9]+.[0-9]+-*" # Push events to matching v*, i.e. v1.0-rc1, v20.15.10-alpha1
 
 env:
   registry: docker.io
-  repository: finschia/finschianode
+  DOCKER_REPOSITORY: finschia/finschianode
```

Comment:
Why should we provide multiple docker images for the same binary?
Can you elaborate the background?

Reply:
Because the optimal version for each system  runs faster. 
For example, in my case, I use M1 mac. So when I use `linux/arm64`, the docker image more fast then `linux/amd64`.

Reply:
The point is not about architectures, but about the base image. Architecture support is important feature.
You've added 3 base images. But I think one image would be sufficient.

Reply:
Our current system and test are fitted with `alpine` base image. But as you said, `alpine` has an elementf of insecurity, and I also think it's good to use `debian11` when we use just `fnsad` cli without any compile in any platform, even windows. So I added `debian11` and `debian11-noroot` base image. Do you think it is no need?

Reply:
But I also think other base image is no need. Because alpine can do it and replace it. 
I'll leave `alpine` and delete the rest. Thank you.

Reply:
> But as you said, `alpine` has an elementf of insecurity,

I've not said about any security concerns, please check the corresponding docker hub (https://hub.docker.com/_/golang). So I think we should have a discussion about it.

> it's good to use `debian11` when we use just `fnsad` cli without any compile in any platform, even windows.

Technically, any "good" base image would allow running the binary. One of the reasons of using container is encapsulating the running environment, so it's OK to use shared library on it. However, combining with the cosmovisor, it would be better to make it static. Then, practically, any linux kernel based container would be sufficient.

TL;DR
1. It doesn't matter whether the executable is static or not, in containers.
2. However, with the cosmovisor, we should provide the static executables.

In my opinion, we should build the static executable, and our containers should contain that executable (for maintenance reason).

> So I added `debian11` and `debian11-noroot` base image. Do you think it is no need?

Can you elaborate the differences of the both images?

Reply:
`Distroless` debian11 image does not contain package managers and shell script, and we can just contain `fnsad` and runtime dependencies. So it is small size and I think we can generate more secure image in terms of security.
So I add this base image in our docker image. 

https://github.com/GoogleContainerTools/distroless

Reply:
OK. I think we can stick to alpine for now. Using golang:alpine seems a little bit tricky, but also OK for now because it's working. We may address it when the related issue arises.